nomad/deployment_endpoint.go

```diff
@@ -52,14 +52,15 @@ func (d *Deployment) GetDeployment(args *structs.DeploymentSpecificRequest,
 				return err
 			}
 
+			// Re-check namespace in case it differs from request.
+			if out != nil && !allowNsOp(aclObj, out.Namespace) {
+				// 404
```

Comment:
plain 404 comment is a bit cryptic for a fresh eye IMO. Don't have constructive feedback - i'd probably suggest removing it; or expanding it a bit (e.g. "conceal deployments if not authorized to view") or something like that.